SignalServiceKit/src/Storage/OWSOrphanedDataCleaner.m

```diff
@@ -129,22 +131,30 @@ + (void)auditAndCleanup:(BOOL)shouldCleanup completion:(void (^_Nullable)(void))
                                                           addObjectsFromArray:quotedMessage
                                                                                   .thumbnailAttachmentStreamIds];
                                                   }
+
+                                                  OWSContact *_Nullable contactShare = message.contactShare;
+                                                  if (contactShare && contactShare.avatarAttachmentId) {
+                                                      [contactShareAvatarAttachmentIds
+                                                          addObject:contactShare.avatarAttachmentId];
+                                                  }
                                               }];
     }];
 
-    CleanupLogDebug(@"%@ attachmentIds: %zd", self.logTag, attachmentIds.count);
-    CleanupLogDebug(@"%@ messageAttachmentIds: %zd", self.logTag, messageAttachmentIds.count);
-    CleanupLogDebug(@"%@ quotedReplyThumbnailAttachmentIds: %zd", self.logTag, quotedReplyThumbnailAttachmentIds.count);
+    CleanupLogDebug(@"%@ attachmentIds: %tu", self.logTag, attachmentIds.count);
+    CleanupLogDebug(@"%@ messageAttachmentIds: %tu", self.logTag, messageAttachmentIds.count);
+    CleanupLogDebug(@"%@ quotedReplyThumbnailAttachmentIds: %tu", self.logTag, quotedReplyThumbnailAttachmentIds.count);
+    CleanupLogDebug(@"%@ contactShareAvatarAttachmentIds: %tu", self.logTag, contactShareAvatarAttachmentIds.count);
 
     NSMutableSet<NSString *> *orphanAttachmentIds = [attachmentIds mutableCopy];
     [orphanAttachmentIds minusSet:messageAttachmentIds];
     [orphanAttachmentIds minusSet:quotedReplyThumbnailAttachmentIds];
+    [orphanAttachmentIds minusSet:contactShareAvatarAttachmentIds];
     NSMutableSet<NSString *> *missingAttachmentIds = [messageAttachmentIds mutableCopy];
     [missingAttachmentIds minusSet:attachmentIds];
 
-    CleanupLogDebug(@"%@ orphan attachmentIds: %zd", self.logTag, orphanAttachmentIds.count);
-    CleanupLogDebug(@"%@ missing attachmentIds: %zd", self.logTag, missingAttachmentIds.count);
-    CleanupLogDebug(@"%@ orphan interactions: %zd", self.logTag, orphanInteractionIds.count);
+    CleanupLogDebug(@"%@ orphan attachmentIds: %tu", self.logTag, orphanAttachmentIds.count);
+    CleanupLogDebug(@"%@ missing attachmentIds: %tu", self.logTag, missingAttachmentIds.count);
+    CleanupLogDebug(@"%@ orphan interactions: %tu", self.logTag, orphanInteractionIds.count);
```

Comment:
Is `%tu` right?

> t | Length modifier specifying that a following d, o, u, x, or X conversion specifier applies to a ptrdiff_t.

from https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/Strings/Articles/formatSpecifiers.html

Reply:
Yeah, it happens to work on 32 and 64 bit, but it seems like a hack. The problem with the existing zd, is that although there's no warning, it's the wrong type - (i.e. very large unsigned ints are shown as negative.)

I've replaced all these usages with %lu and an explicit cast to `(unsigned long)` which is the party line.